src/modules/household/components/ManageHousehold.tsx

```diff
@@ -49,171 +49,192 @@ interface Props {
+  const [searchOpen, setSearchOpen] = useState(!householdId);
+  const [previousMembersOpen, setPreviousMembersOpen] = useState(false);
 
-  // Fetch members to show in "previously associated" table
-  const [recentMembers, recentMembersLoading] = useRecentHouseholdMembers(
-    currentDashboardClientId
+  const onSuccess = useCallback(
+    (newHouseholdId: string) => {
+      if (!householdId && onFirstMemberAdded) {
+        onFirstMemberAdded(newHouseholdId);
+      }
+      setSearchOpen(false);
+      setSearchInput(undefined);
+      setPreviousMembersOpen(false);
+    },
+    [householdId, onFirstMemberAdded]
   );
-  const [recentEligibleMembers, setRecentEligibleMembers] =
-    useState<RecentHouseholdMember[]>();
-
-  useEffect(() => {
-    if (!currentDashboardClientId) return;
-    if (loading || !recentMembers) return;
+  const { addToEnrollmentColumns, refetchHousehold, household, loading } =
+    useAddToHouseholdColumns({
+      householdId,
+      project,
+      onSuccess,
+    });
 
-    const hc = household?.householdClients || [];
-    const currentMembersMap = new Set(hc.map((c) => c.client.id));
-
-    setRecentEligibleMembers(
-      recentMembers.filter(({ client }) => !currentMembersMap.has(client.id))
-    );
-  }, [currentDashboardClientId, recentMembers, household, loading]);
+  // Fetch members to show in "previously associated" table
+  const {
+    previouslyAssociatedMembers,
+    previouslyAssociatedMembersDescription,
+  } = usePreviouslyAssociatedMembers(household);
 
-  useScrollToHash(loading || recentMembersLoading);
   const isMobile = useIsMobile();
 
-  const columns = useMemo(() => {
-    const defaults: ColumnDef<
-      ClientSearchResultFieldsFragment | RecentHouseholdMember
-    >[] = [CLIENT_COLUMNS.name, CLIENT_COLUMNS.dobAge];
-    if (isMobile) {
-      // On mobile, show enrollment button right next to the client name so user
-      // doesn't have to scroll to the right.
-      defaults.splice(1, 0, ...addToEnrollmentColumns);
-    } else {
-      defaults.push(...addToEnrollmentColumns);
-    }
-    if (globalFeatureFlags?.mciId) {
-      return [
-        externalIdColumn(ExternalIdentifierType.MciId, 'MCI ID'),
-        ...defaults,
-      ];
-    }
-    return defaults;
-  }, [addToEnrollmentColumns, isMobile, globalFeatureFlags?.mciId]);
+  const searchResultColumns: ColumnDef<
+    ClientSearchResultFieldsFragment | RecentHouseholdMember
+  >[] = useMemo(() => {
+    return [
+      { ...CLIENT_COLUMNS.name, sticky: 'left', width: '25%' }, // why does sticky make it so wide?
+      ...(globalFeatureFlags?.mciId
+        ? [externalIdColumn(ExternalIdentifierType.MciId, 'MCI ID')]
+        : []),
+      // On mobile, show enrollment button right next to the client name so user.
+      // This needs fixing, ideally we could use a sticky column but it doesn't work as expected on mobile
+      ...(isMobile ? addToEnrollmentColumns : []),
```

Comment:
+

Reply:
I don't think that we have an open issue for this, but I'm happy to create one!

Is this PR ready for review in general? The functionality looks amazing from what I have poked at, but I saw that it is still in draft so I haven't begun a proper review. Let me know if I should.

Reply:
Created https://github.com/open-path/Green-River/issues/7338

Reply:
Thank you! Yes it's ready for review, I'll mark it now. Sorry it's such a monster, happy to walk through some changes together if that would be helpful!